### PR TITLE
Basic travel management

### PR DESCRIPTION
**Pull request by kberzinch**

### CODE REVIEW

File: app/Http/Controllers/DashboardController.php [hunk 1]
```diff
@@ -89,6 +89,24 @@ public function index(Request $request): View
...
+        $needTravelDocuments = false;
+        $needTravelPayment = false;
+        $travelName = '';
+
+        if (null !== $assignment) {
+            if (! $assignment->documents_received && null !== $assignment->travel->documents_required) {
+                $needTravelDocuments = true;
+            }
+
+            if (! $assignment->is_paid) {
```
Comment (evan10s) [inline app/Http/Controllers/DashboardController.php:103]: You can make this assignment one line

File: database/migrations/2021_05_15_121109_create_travel_table.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    public function up(): void
+    {
+        Schema::create('travel', static function (Blueprint $table): void {
+            $table->id();
+            $table->string('name');
+            $table->string('destination');
+            $table->unsignedInteger('primary_contact_user_id');
+            $table->date('departure_date');
+            $table->date('return_date');
+            $table->unsignedInteger('fee_amount');
```
Comment (evan10s) [inline database/migrations/2021_05_15_121109_create_travel_table.php:23]: Should probably allow decimal places here since it's currency

Comment (kberzinch) [reply database/migrations/2021_05_15_121109_create_travel_table.php:23]: I thought about that, but in practice I don't think that is useful.

File: app/Nova/Travel.php [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+            Markdown::make('Included with Fee')
+                ->required()
+                ->rules('required')
+                ->help('Describe what costs will be covered by RoboJackets.'),
+
+            Markdown::make('Not Included with Fee')
+                ->help('Describe what costs are anticipated to be covered by members themselves.'),
+
+            Markdown::make('Documents Required')
+                ->help('Describe what documents will be required to travel.'),
```
Comment (JoshOldenburg) [inline app/Nova/Travel.php:97]: Should we give some suggestions in the help?

File: resources/views/layouts/app.blade.php [hunk 4]
```diff
@@ -28,6 +28,12 @@
             </li>
             @endif
 
+            @if (auth()->user() && auth()->user()->assignments()->count() > 0)
+            <li class="nav-item {{ $request->is('travel*') ? 'active' : '' }}">
+              <a class="nav-link" href="{{ route('travel.index') }}">Travel<sub><small>&nbspbeta</small></sub></a>
```
Comment (JoshOldenburg) [inline resources/views/layouts/app.blade.php:33]: Nice "beta" warning

File: resources/views/travel/index.blade.php [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+                  <td>${{ $travel->fee_amount }}</td>
+                </tr>
+                <tr>
+                  <th scope="row">Costs Included</th>
+                  <td>@markdown($travel->included_with_fee)</td>
+                </tr>
+                @if($travel->not_included_with_fee)
+                <tr>
+                  <th scope="row">Not Included</th>
+                  <td>@markdown($travel->not_included_with_fee)</td>
```
Comment (JoshOldenburg) [inline resources/views/travel/index.blade.php:49]: Is this injection-safe?

**Review by JoshOldenburg — CHANGES_REQUESTED** [review]
Looking pretty good, but a few minor comments.

File: resources/views/travel/index.blade.php (hunk 5 above)
Comment (kberzinch) [reply resources/views/travel/index.blade.php:49]: I think we looked at this for membership agreements. You can run arbitrary code in browser but not on the server.

It's limited to PM+ so I think it should be fine.

**Review by evan10s — APPROVED** [review]
Code-wise seems fine

**Review by JoshOldenburg — APPROVED** [review]

**Pull request merged**